airflow/metrics/datadog_logger.py

```diff
@@ -51,7 +55,33 @@
     )
 
 
-class SafeDogStatsdLogger:
+def prepare_stat_with_tags(fn: T) -> T:
+    """Prepare tags and stat."""
+
+    @wraps(fn)
+    def wrapper(self, stat: str | None = None, *args, tags: dict[str, str] | None = None, **kwargs):
+        stat = stat or ""
```

Comment:
Nit: the `stat` has been renamed to `metric_name` elsewhere, so lets do it too.

```suggestion
    def wrapper(self, metric_name: str | None = None, *args, tags: dict[str, str] | None = None, **kwargs):
        metric_name = metric_name or ""
```